src/GitVersion.Core/Helpers/ValueFormatter.cs

```diff
@@ -0,0 +1,36 @@
+namespace GitVersion.Helpers;
+
+internal static class ValueFormatter
+{
+    private static readonly List<IValueFormatter> _formatters =
+    [
+        new StringFormatter(),
+        new FormattableFormatter(),
+        new NumericFormatter(),
+        new DateFormatter()
+    ];
+
+    public static bool TryFormat(object? value, string format, out string result)
+    {
+        result = string.Empty;
+
+        if (value is null)
+        {
+            return false;
+        }
+
+        foreach (var formatter in _formatters.OrderBy(f => f.Priority))
```

Comment:
…instead we use the `this.` qualifier to avoid ambiguity between fields and local variables.

```suggestion
        foreach (var formatter in this.formatters.OrderBy(f => f.Priority))
```